wordpress-mobile/WordPress-iOS
# Decoupling Authentication Flow: Mark VI

### Details:
In this PR we're removing `AccountHelper` usage in the Auth ViewControllers. We're replacing this with a `WordPressAuthenticatorDelegate` method, that will indicate if the Auth UI is dismissable, or not.

### Testing:
1. Log into a dotcom / self hosted site
2. Open `My Site` (and hit Back to view the Sites List)
3. Press over the `+` button at the top right corner
4. Press `Add a Self Hosted Site` / `Create WPCOM Site)
5. Verify a `Cancel` button shows up at the top left corner, and after pressing it, the modal UI is dismissed.

Needs Review: @nheagy
